common/src/main/java/hardcorequesting/common/quests/task/CheckBoxTask.java

```diff
@@ -0,0 +1,63 @@
+package hardcorequesting.common.quests.task;
+
+import com.google.gson.JsonObject;
+import hardcorequesting.common.io.adapter.Adapter;
+import hardcorequesting.common.quests.Quest;
+import hardcorequesting.common.quests.data.TaskData;
+import hardcorequesting.common.team.Team;
+import net.minecraft.world.entity.player.Player;
+
+import java.util.UUID;
+
+public class CheckBoxTask extends QuestTask<TaskData> {
+    public CheckBoxTask(Quest parent) {
+        super(TaskType.CHECKBOX, TaskData.class, parent);
+    }
+
+    @Override
+    public TaskData newQuestData() {
+        return new TaskData();
+    }
+
+    @Override
+    public void onUpdate(Player player) {
+        if (getData(player).completed) {
+            // completed flag was set by client via update packet
```

Comment:
I am not confident that this comment holds true. Can you elaborate on where/how this flag is being set?

Reply:
This is the code that sets it: https://github.com/lorddusk/HQM/pull/628/files#diff-1ebcf341ba9926e5c3b9b3a25324aa214a324808a09ee45d53e388767cf5651dR31

Reply:
However is that not being set on the logical client-side?

Reply:
Kind of? The server is doing the actual work of completing the task.

Reply:
Yes, but the completed field isn't normally synced from client to server, right? So you wouldn't be able to reliably read the value server-side that you're setting client-side.

Reply:
Ah hell, I'll just make a custom packet for it